egi2bids/mff2bids.py

```diff
@@ -47,58 +57,115 @@
 # fmt: on
 
 
+def _extract_folder(
+    file: Union[str, Path], dir_: Union[str, Path] = None
+) -> Path:
+    """Extract a .mff compressed folder to its original form."""
+    # check paths and file extension
+    file = _ensure_path(dir_, must_exist=True)
+    ext = file.suffix
+    _check_value(ext, (".tar", ".zip", ".mff"), "extension")
+    dir_ = Path.cwd() if dir_ is None else dir_
+    dir_ = _ensure_path(dir_, must_exist=True)
+    # open the archive if needed
+    archive_readers = {
+        ".tar": tarfile.open,
+        ".zip": zipfile.ZipFile,
+    }
+    if ext in (".tar", ".zip"):
+        logger.info("Extracting '%s' archive %s to %s.", ext, file, dir_)
+        with archive_readers[ext](file, "r") as archive:
+            archive.extractall(dir_)
+        for root, dirs, _ in os.walk(dir):
+            if "Contents" in dirs:
+                logger.info("MFF file found in %s", root)
+                return Path(root)
+        else:
+            raise (f"The '{ext}' archive does not contain a 'Content' folder.")
+
+    elif ext == ".mff":
+        return file
```

Comment:
@vferat I changed the structure of this function, with something a bit more DRY.